src/pegjs/floip.pegjs

```diff
@@ -227,13 +229,18 @@ Expression_Types = Escaped_Identifier / Logic / Math / Concatenation / Function
  * Function looks like @(SOME_METHOD(argument, argument...))
  * Functions always wrap their arguments in parens.
  */
-Function = call:$valid_expression_characters* OpenParen args:( Function_Args* ) CloseParen {
-  return method(call, args, location())
+Function = call:$valid_expression_characters* OpenParen args:( Function_Args* ) CloseParen chain:Chain_Function* {
+  return method(call, args, chain, location())
   /** <?php
-    return call_user_func_array($this->_method, [$call, $args]);
+    return call_user_func_array($this->_method, [$call, $args, $chain]);
   ?> **/
   }
 
+Chain_Function = '.' call:$valid_expression_characters+ {
+  return call
+  /** <?php return $call; ?> **/
```

Comment:
I think we should delete commented code.

Reply:
This is _special_ commented code though. By commenting the PHP code, we can use the same grammar file for both JS and PHP generator output.
See [this](https://github.com/nylen/phpegjs#grammar-syntax-and-semantics)